cloudstack/HostService.go

```diff
@@ -357,7 +357,7 @@ type AddBaremetalHostResponse struct {
 	Jobstatus                        int                                `json:"jobstatus"`
 	Lastannotated                    string                             `json:"lastannotated"`
 	Lastpinged                       string                             `json:"lastpinged"`
-	Managementserverid               string                             `json:"managementserverid"`
+	Managementserverid               CSLong                             `json:"managementserverid"`
```

Comment:
Could something like this work?
```suggestion
	Managementserverid               string                             `json:"managementserverid,string"`
```

Reply:
I'm not familiar with this construction but from my understanding it won't work.
- first: [encoding/json documentation](https://pkg.go.dev/encoding/json#Marshal) states that this tags use use to tell the library to store the value as a json-encoded string. It cannot be use to read an json-long as a string
- second: I created a test program on golang's playgournd that shows that, indeed, this construction wont work. https://play.golang.org/p/iS3FVwp7mik

Reply:
```go
package main

import (
	"fmt"
	"encoding/json"
)

type test struct {
  Test string `json:"test,string"`
}

func main() {
	val := `{"test": 12564564}`
	t := test{}
	err := json.Unmarshal([]byte(val), &t)
	if err != nil {
		fmt.Printf("error: %s\n", err)
	} else {
		fmt.Printf("%#v\n", t)
	}
}
```
`error: json: invalid use of ,string struct tag, trying to unmarshal unquoted value into string`